coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt

```diff
@@ -128,29 +127,29 @@ class BlockCreationMonitor(
           SafeFuture.COMPLETE
         } else {
           getNetNextSafeBlock()
-            .thenCompose { payload ->
-              if (payload != null) {
-                if (payload.parentHash == expectedParentBlockHash.get()) {
-                  notifyListener(payload)
+            .thenCompose { block ->
+              if (block != null) {
+                if (block.parentHash.contentEquals(expectedParentBlockHash.get())) {
+                  notifyListener(block)
                     .whenSuccess {
                       log.debug(
                         "updating nexBlockNumberToFetch from {} --> {}",
                         _nexBlockNumberToFetch.get(),
                         _nexBlockNumberToFetch.incrementAndGet()
                       )
-                      expectedParentBlockHash.set(payload.blockHash)
+                      expectedParentBlockHash.set(block.hash)
                     }
                 } else {
                   reorgDetected.set(true)
                   log.error(
                     "Shooting down conflation poller, " +
                       "chain reorg detected: block { blockNumber={} hash={} parentHash={} } should have parentHash={}",
-                    payload.blockNumber.longValue(),
-                    payload.blockHash.toHexString().subSequence(0, 8),
-                    payload.parentHash.toHexString().subSequence(0, 8),
-                    expectedParentBlockHash.get().toHexString().subSequence(0, 8)
+                    block.number,
+                    block.hash.encodeHex(),
+                    block.parentHash.encodeHex(),
+                    expectedParentBlockHash.get().encodeHex().subSequence(0, 8)
```

Comment:
```suggestion
                    expectedParentBlockHash.get().encodeHex()
```
Maybe we should remove the truncation here as well for consistency?